job-runner/webapp/server/classes/UpdateCbioData.js

```diff
@@ -0,0 +1,155 @@
+    return {
+      data_set_id: data_set_id,
+      data_set_name: samplesAndName.data_set_name,
+      sample_labels: samplesAndName.sample_labels,
+
+      // I think we can fake this
+      unfiltered_sample_count: 1,
+    };
+  });
+
+  //var comboSampleGroupId = SampleGroups.insert({
+  //  name: "temp - created in UpdateCbioData to call an adapter",
+  //  version: 1,
+  //  data_sets: comboSampleGroupDataSets,
+  //  value_type: group.value_type,
+
+    // invisible
+  //  collaborations: [],
+  //});
+
+  // star the promise chain: woohoo!
+
+  var self = this;
+  var deferred = Q.defer();
+
+  // define up here so as to be available throughout promise chain (so that
+  // we can skip a .then block)
+  var geneSetGroupPath;
+  var expressionDataPath = path.join(workDir, "data_expression.txt");
+  var logPath = path.join(workDir, "*.log");
+  var associated_object = {
+    collection_name: "SampleGroups",
+    mongo_id: group._id,
+  };
+  var form_id = this.job.args.form_id;
+
+  Q.all([
+      // write mongo data to files
+
+      // expression data to a file for use in Limma
+      spawnCommand(getSetting("genomic_expression_export"), [
+        "--sample_group_id", group._id,
+        "--cbio",
+        "--uq-sample-labels",
+      ], workDir, { stdoutPath: expressionDataPath }),
+      // phenotype file for cbio importer
+      spawnCommand(getSetting("clinical_export"), [
+        "--sample_group_id",
+        this.job.args.sample_group_id,
+        "--form_id",
+        this.job.args.form_id,
+        "--work-dir",
+
+        workDir,
+      ], workDir),
+    ])
+    .then(function (spawnResults)           {
+      console.log("done writing files");
+
+      _.each(spawnResults, function (result) {
+        if (result.exitCode !== 0) {
+          throw "Problem writing files to disk.";
+        }
+      });
+
+      // save the file paths... order maters for spawnResults
+      // (the order depends on the order of `spawnCommand`s in `Q.all`)
+      var clinicalPath = spawnResults[1].stdoutPath;
+
+      // run CbioImporter
+      return spawnCommand(getSetting("cbio_importer_path"), [
+        "-c import-study",
+        "-jar",
+        getSetting("cbio_core_jar_path"),
+        "-s",
+        workDir,
+      ], workDir );
+    })
+    .then(Meteor.bindEnvironment(function (cbioImportResult) {
+      if (cbioImportResult.exitCode !== 0) {
+        throw "Problem running cbio importer";
+      }
+      function printError (err) {
+         if (err) {
+           console.log("error creating blob:", err);
+         }
+       }
+      var meta = { form_id: form_id, cbio_import: dataSetName};
+      console.log('importing into cbio form', form_id, dataSetName, associated_object)
+      Blobs2.create(cbioImportResult.stdoutPath, associated_object, meta, printError);
+
+     deferred.resolve(cbioImportResult);
+   }, deferred.reject))
+    .catch(Meteor.bindEnvironment(function (reason) {
+
+      deferred.reject(reason);
```

Comment:
This is more of a general thing, but how often are we planning to reload the cBio data? Do we need a simple UI to track the import jobs that's not linked to anywhere? (Similar to [the widgets demo page](https://medbook.io/widgets)) This would allow someone to load new data into cBio easily without having access to the database.